contracts/libraries/TokenAuthorizationCache.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+/// @title a library to store spenders' authorization over v3 position token ids in transient storage
+/// @dev this library implements the equivalent of a mapping, as transient storage can only be accessed in assembly
+library TokenAuthorizationCache {
```

Comment:
i've been thinking something like:
1. I approve a hook contract for all of my v3 position nfts by calling setApprovalForAll()
2. I call the UR and execute erc721permit (to permit the UR), decreaseLiquidity, swapV4, collect
3. In the swapV4 call, it calls out to the hook contract which transfers my position nft to someone else, making me not the owner anymore
4. Collect still works since my authorization is cached and I receive the tokens owed even though I am not the owner anymore

Reply:
> Collect still works since my authorization is cached

But would collect actually still work? I feel like when the actual `collect` call is executed, postiion manager will check "is UR approved for the user". And the NEW user has to have also approved UR for this to happen.

But yes i think if this new user has also approved the UR then it would likely go through 🙃